core/src/main/java/io/fabric8/maven/core/util/JibBuildServiceUtil.java

```diff
@@ -0,0 +1,237 @@
+            entrypointList = buildConfiguration.getEntryPoint().asStrings();
+        }
+
+        buildImage(fromImage, targetImage, envMap, credential, portSet, fatJar, entrypointList, targetDir, outputDir, log);
+    }
+
+    /**
+     * Builds a container image using Jib from all the following parameters:
+     *
+     * @param baseImage
+     * @param targetImage
+     * @param envMap
+     * @param credential
+     * @param portSet
+     * @param fatJar
+     * @param entrypointList
+     * @param targetDir
+     * @param outputDir
+     * @param log
+     * @return
+     * @throws InvalidImageReferenceException
+     */
+    protected static JibContainer buildImage(String baseImage, String targetImage, Map<String, String> envMap, Credential credential, Set<Port> portSet, Path fatJar, List<String> entrypointList, String targetDir, String outputDir, Logger log) throws InvalidImageReferenceException {
+        String username = "";
+        String password = "";
+
+        JibContainerBuilder contBuild = Jib.from(baseImage);
+
+        if (envMap != null) {
+            contBuild = contBuild.setEnvironment(envMap);
+        }
+
+        if (portSet != null) {
+            contBuild = contBuild.setExposedPorts(portSet);
+        }
+
+        if (fatJar != null) {
+            String fatJarName = fatJar.getFileName().toString();
+            String jarPath = targetDir + "/" + (fatJarName.isEmpty() ? DEFAULT_JAR_NAME: fatJarName);
+            contBuild = contBuild
+                    .addLayer(LayerConfiguration.builder().addEntry(fatJar, AbsoluteUnixPath.get(jarPath)).build());
+        }
+
+        if(!entrypointList.isEmpty()) {
+            contBuild = contBuild.setEntrypoint(entrypointList);
+        }
+
+        if (credential != null) {
+            username = credential.getUsername();
+            password = credential.getPassword();
+
+            if (targetImage.contains(DEFAULT_USER_NAME)) {
+                targetImage = targetImage.replaceFirst(DEFAULT_USER_NAME, username + "/");
+            }
+        }
+
+        RegistryImage registryImage = RegistryImage.named(targetImage).addCredential(username, password);
+        String imageTarName = ImageReference.parse(targetImage).getRepository().concat(".tar");
+        TarImage tarImage = TarImage.named(targetImage).saveTo(Paths.get(outputDir + "/" + imageTarName));
+        try {
+            JibContainer jibContainer = buildContainer(contBuild, registryImage, log);
+            log.info("Image %s successfully built and pushed.", targetImage);
+            return jibContainer;
+        } catch (RegistryException re) {
+            log.warn("Registry Exception occured : %s", re.getMessage());
+            log.warn("Credentials are probably either not configured or are incorrect.");
+            log.info("Building Image Tarball at %s.", imageTarName);
+            JibContainer jibContainer =  buildContainer(contBuild, tarImage, log, false);
```

Comment:
@dev-gaur : Can we make this a flag so that the user can also override in case he wants an offline build?